### PR TITLE
Add Permission, update the Authorizer

### PR DESCRIPTION
**Pull request by kingiw**
1. add class `Profiles` to manage the accessibility.
2. new initialization method for `Authorizer`: Initializing the authorizer without specifying an endpoint. In this way, user need to pass the profiles to the authorizer manually via `setProfiles`.
3. add `can`, `cannot`, `canAll`, `canAny` to verify the permissions.

### CODE REVIEW

**Comment by kingiw** [comment]
@hsluoyz @nodece Please review.

File: src/profiles.ts [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+            } 
+            tmp[jsonObj[act]].push(act);
+        }
+        for (const obj in tmp) {
+            this.objActData.set(obj, tmp[obj] as Array<string>);
+        }
+
+    }
+
+    public getProfilesJson() : {[key: string]: string[]} {
```
Comment (nodece) [inline src/profiles.ts:35]: You should define `{[key: string]: string[]}` as type.

File: src/__test__/profiles.test.ts [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+    const policyExample = {
+        'read': ['data1', 'data2'],
+        'write': ['data2']
+    }
+    const profiles = new Profiles();
+
+    // From https://stackoverflow.com/a/57428906/7409994
+    const expectArrayEquivalence = <T>(actual: T[], expected: T[]) => {
+        expect(actual).toEqual(expect.arrayContaining(expected));
+        expect(expected).toEqual(expect.arrayContaining(actual));
```
Comment (nodece) [inline src/__test__/profiles.test.ts:12]: Try sorting the actual and expected array, somethings like
```
expect(actual.sort()).toEqual(expected.sort());
```

File: src/profiles.ts [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+
+    public constructor() {
+        this.actObjData = new Map<string, Array<string>>();
+        this.objActData = new Map<string, Array<string>>();
+    }
+    
+    /*
+        Load the profiles from a JSON string
+    */
+    public loadFromString(s : string) : void {
```
Comment (nodece) [inline src/profiles.ts:13]: Add `public load(data: object) : void` to this is better.

File: src/__test__/index.test.ts [hunk 4]
```diff
@@ -13,9 +13,34 @@ test('Mock functions', async () => {
...
     // expect(mockedAxios.get('http://localhost:4000/api/casbin?casbin_subject=alice')).toMatchObject(respObj);
 });
+
+test('Authorizer', () => {
+    const profilesObj = {
+        read: ['data1', 'data2', 'data3'],
+        write: ['data1']
+    }
+    const authorizer = new Authorizer();
+    authorizer.setProfiles(JSON.stringify(profilesObj));
```
Comment (nodece) [inline src/__test__/index.test.ts:29]: I prefer using `authorizer.setProfiles(profilesObj);`

File: src/profiles.ts (hunk 1 above)
Comment (hsluoyz) [reply src/profiles.ts:35]: Yes. We should define a type if it's complex.

File: src/profiles.ts (hunk 3 above)
Comment (kingiw) [reply src/profiles.ts:13]: Do you mean to add an interface supported for objects in any type like:
```
public load(data: object) : void {
  if (data instanceof String) {
    // Load from String
  } else if (data instanceof OtherType) {
    // ...
  }
}
```

File: src/__test__/index.test.ts (hunk 4 above)
Comment (kingiw) [reply src/__test__/index.test.ts:29]: Actually `setProfiles` (now it is renamed as `setPermission`) accept a string in JSON format as the input. The `profilesObj` here is a JS object, so I stringify it before passing it to the function.

**Pull request merged**